egui_extras/src/table.rs

```diff
@@ -0,0 +1,280 @@
+        self.sizing.add_size(width);
+        self
+    }
+
+    /// Add size hint for column [count] times
+    pub fn columns(mut self, size: Size, count: usize) -> Self {
+        for _ in 0..count {
+            self.sizing.add_size(size.clone());
+        }
+        self
+    }
+
+    /// Create a header row which always stays visible and at the top
+    pub fn header(self, height: f32, header: impl FnOnce(TableRow<'_, '_>)) -> Table<'a> {
+        let widths = self.sizing.into_lengths(
+            self.ui.available_rect_before_wrap().width() - 2.0 * self.padding.outer,
+            self.padding.inner,
+        );
+        let ui = self.ui;
+        {
+            let mut layout = Layout::new(ui, self.padding.clone(), LineDirection::TopToBottom);
+            {
+                let row = TableRow {
+                    layout: &mut layout,
+                    widths: widths.clone(),
+                    striped: false,
+                    height,
+                    clicked: false,
+                };
+                header(row);
+            }
+        }
+
+        Table {
+            ui,
+            padding: self.padding,
+            widths,
+            scroll: self.scroll,
+            striped: self.striped,
+        }
+    }
+
+    /// Create table body without a header row
+    pub fn body<F>(self, body: F)
+    where
+        F: for<'b> FnOnce(TableBody<'b>),
+    {
+        let widths = self.sizing.into_lengths(
+            self.ui.available_rect_before_wrap().width() - 2.0 * self.padding.outer,
+            self.padding.inner,
+        );
+
+        Table {
+            ui: self.ui,
+            padding: self.padding,
+            widths,
+            scroll: self.scroll,
+            striped: self.striped,
+        }
+        .body(body)
+    }
+}
+
+pub struct Table<'a> {
+    ui: &'a mut Ui,
+    padding: Padding,
+    widths: Vec<f32>,
+    scroll: bool,
+    striped: bool,
+}
+
+impl<'a> Table<'a> {
+    /// Create table body after adding a header row
+    pub fn body<F>(self, body: F)
+    where
+        F: for<'b> FnOnce(TableBody<'b>),
+    {
+        let padding = self.padding;
+        let ui = self.ui;
+        let widths = self.widths;
+        let striped = self.striped;
+        let start_y = ui.available_rect_before_wrap().top();
+        let end_y = ui.available_rect_before_wrap().bottom();
+
+        egui::ScrollArea::new([false, self.scroll]).show(ui, move |ui| {
+            let layout = Layout::new(ui, padding, LineDirection::TopToBottom);
+
+            body(TableBody {
+                layout,
+                widths,
+                striped,
+                odd: true,
+                start_y,
+                end_y,
+            });
+        });
+    }
+}
+
+pub struct TableBody<'a> {
+    layout: Layout<'a>,
+    widths: Vec<f32>,
+    striped: bool,
+    odd: bool,
+    start_y: f32,
+    end_y: f32,
+}
+
+impl<'a> TableBody<'a> {
+    /// Add rows with same height
```

Comment:
```suggestion
    /// Add rows with same height.
    ///
```

Should the height include the spacing?